### DISCUSSION TITLE
Our insights with iOS

### DISCUSSION THREAD

**Discussion by Textfield** [six7/figma-tokens#243] [Q&A]
Hello,

We wanted to test the Figma Tokens plugin with our iOS engineers to see if we can create a .json file to control and update the designs from the .json export directly.

Our engineering teams insight was, since values come as a String they need to be reverse engineered. Especially using math such as $spacing.small * 2 is problematic. Is there some kind of guidance I can guide them with a nice practice?

Thank you!

**Comment by six7** [comment]
I'm working on a tool that takes the raw output of Figma Tokens and transforms it to a resolved json that doesn't contain any math or aliases. This kind of requires you to specify what themes you'd want activated, but that could be a command line flag that you pass and could be automated.

e.g. for each push to GH you define in a GH Action that you want to create one file for light theme, one for dark and maybe some others. You could then use that json to maybe combine them or transform them with style dictionary.

If you're looking for a solution right now you'd have to build it yourself :/ But I'll keep you updated as I want to release this together with the GitHub sync.

**Comment by six7** [comment]
I just published a package that does the transformation for you and outputs a json that has no aliases or math.

https://www.npmjs.com/package/token-transformer

It's part of the GitHub sync beta that I'll launch in a bit.

Here's information on how to use this:
https://figma-tokens-docs-git-git-sync-beta-six7.vercel.app/sync/github

And the part relevant to only development would be https://figma-tokens-docs-git-git-sync-beta-six7.vercel.app/sync/github#7-how-to-use-these-in-development
